src/test/java/seedu/dengue/logic/commands/DeleteCommandTest.java

```diff
@@ -197,6 +206,285 @@ public void execute_invalidMultiIndexFilteredList_throwsCommandException() {
+
+        DeleteCommand deleteCommand = new DeleteCommand(predicate);
+
+        String expectedMessage = String.format(DeleteCommand.MESSAGE_DELETE_RANGE_SUCCESS, 1,
+                dateRange.getStart(), dateRange.getEnd());
+
+        ModelManager expectedModel = new ModelManager(model.getDengueHotspotTracker(), new UserPrefs());
+        expectedModel.deletePerson(FIONA);
+
+        assertCommandSuccess(deleteCommand, model, expectedMessage, expectedModel);
+    }
+
+    @Test
+    public void execute_partialRangeEndGivenFilteredList_success() {
+
+        Optional<SubPostal> emptySubPostal = Optional.empty();
+        Optional<Name> emptyName = Optional.empty();
+        Optional<Age> emptyAge = Optional.empty();
+        Optional<Date> emptyDate = Optional.empty();
+        Set<Variant> emptyVariants = new HashSet<>();
+        Range<Date> dateRange = ContinuousData.generateRange(
+                new StartDate(Optional.empty()),
+                new EndDate(Optional.of(new Date(VALID_DATE_FIONA))));
+        Range<Age> emptyAgeRange = ContinuousData.generateRange(new StartAge(emptyAge), new EndAge(emptyAge));
+        FindPredicate predicate = new FindPredicate(
+                emptyName, emptySubPostal, emptyAge, emptyDate, emptyVariants, dateRange, emptyAgeRange);
+
+        ModelManager expectedModel = new ModelManager(model.getDengueHotspotTracker(), new UserPrefs());
+        model.updateFilteredPersonList(predicate);
+        expectedModel.updateFilteredPersonList(predicate);
+
+        DeleteDatePredicate deletePredicate = new DeleteDatePredicate(dateRange);
+
+        DeleteCommand deleteCommand = new DeleteCommand(deletePredicate);
+
+        String expectedMessage = String.format(DeleteCommand.MESSAGE_DELETE_RANGE_SUCCESS, 1,
+                dateRange.getStart(), dateRange.getEnd());
+
+        expectedModel.deletePerson(FIONA);
+        showNoPerson(expectedModel);
+
+        assertCommandSuccess(deleteCommand, model, expectedMessage, expectedModel);
+    }
+
+    @Test
+    public void execute_completeRangeSameDatesUnfilteredList_success() {
+
+        Range<Date> dateRange = ContinuousData.generateRange(
+                new StartDate(Optional.of(new Date(VALID_DATE_ALICE))),
+                new EndDate(Optional.of(new Date(VALID_DATE_ALICE))));
+
+        DeleteDatePredicate predicate = new DeleteDatePredicate(dateRange);
+
+        DeleteCommand deleteCommand = new DeleteCommand(predicate);
+
+        String expectedMessage = String.format(DeleteCommand.MESSAGE_DELETE_RANGE_SUCCESS, 1,
+                dateRange.getStart(), dateRange.getEnd());
+
+        ModelManager expectedModel = new ModelManager(model.getDengueHotspotTracker(), new UserPrefs());
+        expectedModel.deletePerson(ALICE);
+
+        assertCommandSuccess(deleteCommand, model, expectedMessage, expectedModel);
+    }
+
+    @Test
+    public void execute_partialRangeSameDatesFilteredList_success() {
```

Comment:
minor nitpick but maybe rename the beginning of the methods to `executeDelete..` instead of just `execute`